library/ssl_srv.c

```diff
@@ -3142,11 +3139,11 @@ static int ssl_prepare_server_key_exchange( mbedtls_ssl_context *ssl,
             }
         }
         else
-#endif /* MBEDTLS_SSL_PROTO_TLS1_2 */
         {
-            /* C: MD5 + SHA1 */
-            md_alg = MBEDTLS_MD_NONE;
+            MBEDTLS_SSL_DEBUG_MSG( 1, ( "should never happen" ) );
+            return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );
         }
+#endif /* MBEDTLS_SSL_PROTO_TLS1_2 */
```

Comment:
This `#endif` should ideally be between the `else` and the opening `{` on the next line - that way if we ever compile without `PROTO_TLS1_2` (which we currently can't, but that might become possible in the future), we'll still get the internal error path.